cmd/storage/githttp.go

```diff
@@ -339,6 +353,15 @@ func (gh *GitHTTP) serviceHandler(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", fmt.Sprintf("application/x-git-%s-result", service))
 }
 
+func serviceAllowed(service string) error {
```

Comment:
I think this would be best suited to be put into a HTTP middleware and then wrap the necessary handlers with it.
Take a look at the `func cacheForever` on how to do it.